### ISSUE TITLE
[Bug]: completely freezes

### ISSUE THREAD

**Issue by name-used** [matplotlib/matplotlib#28907]
### Bug summary

"When using PyCharm (regardless of the version) in debug mode or starting matplotlib.pyplot.plot in the Python console, the process completely freezes, and I can only force it to end."

### Code for reproduction

```Python
import matplotlib
matplotlib.use('tkagg')
import matplotlib.pyplot as plt
import numpy as np
plt.imshow(np.zeros((10, 10)))
plt.show()
```

### Actual outcome

any version of pycharm

### Expected outcome

nothing

### Matplotlib Version

3.9.*

### Installation

pip

**Comment by jklymak** [comment]
We would need more details than this.  Presumably your install needs to be redone and there is not much we can do.

**Comment by timhoffm** [comment]
"freeze" is hard to debug.

- When you terminate it, do you get a stacktrace?
- Have you tried another backend, say `matplotlib.use("qtagg")`?
- Have you tried to create a new environment? Does it happen there as well?

**Comment by name-used** [comment]
> "freeze" is hard to debug.
> 
> * When you terminate it, do you get a stacktrace?
> * Have you tried another backend, say `matplotlib.use("qtagg")`?
> * Have you tried to create a new environment? Does it happen there as well?

Replay:
> "freeze" is hard to debug.

Yes you are right. It's very hard to debug. I tried many methods to solve this.
Finally I found it happens only when I upgraded matplotlib==3.9.* and use [debug] or [python console] mode with pycharm(any version of this).
So I have to solve this problem by re-install matplotlib==3.8.*.

> * When you terminate it, do you get a stacktrace?

When I terminate it, there is no error message raised in python process.
Messages appear in console are as following:
```
Backend tkagg is interactive backend. Turning interactive mode on.
Process finished with exit code -805306369 (0xCFFFFFFF)
```

> * Have you tried another backend, say `matplotlib.use("qtagg")`?

Yes I do. I tried other backends but useless.
qtagg: I have no idea what this performs. Sometimes it is available, but sometimes performs like tkagg and completely freezes.
qt5agg: I tried it sometimes before but it returns some error message in python process. I tried it again just now and it seems ok.
agg: It is ok but can not help me to show image info intime.
wxagg: It performs like tkagg and completely freezes.

> * Have you tried to create a new environment? Does it happen there as well?

I did not try to create a new enviroment. Actually I coding in both MyPC[win-10] and CompanyServer[Kali-linux-2022] while finding the same error. Both the python versions are 3.9.*.
Due to the characteristics of the problem(Only when I use pycharm[debug] mode or pycharm[python console] mode), I tried update my pycharm with versions from 2022.3.1 to 2024.2.3, but none of them made sense.

**Comment by name-used** [comment]
> We would need more details than this. Presumably your install needs to be redone and there is not much we can do.

Sorry for my late reply. I have provided a detailed explanation of the situation I encountered in my previous response. I am not sure what other information needs to be provided, thank you very much for your reply!

**Comment by jklymak** [comment]
If your problem is just tkagg please check that it works at all in pycharm by running a hello world program.  https://matplotlib.org/stable/users/explain/figure/backends.html#debugging-the-figure-windows-not-showing both in pycharm and in a terminal.

**Comment by jklymak** [comment]
I don't know how pycharm console is meant to work, but can you directly try running the hello-world examples without calling python.exe?  

That 3.8 works, but 3.9 does not is interesting, but that everything works except in pycharm console makes it seem the issue is there, perhaps surfaced by one of our changes.

**Comment by ianthomas23** [comment]
This could well be a duplicate of #28449, meaning that PyCharm need to support the changes in backend registration introduced in Matplotlib 3.9.0.

**Comment by jklymak** [comment]
Agree this seems the same.  

However, I'm confused what is causing this - we became more permissive, but PyCharm expects "TkAgg" and we have allowed the user to call "tkagg"?  That failed before, but now works?  Or is the change that `get_backend` now returns what the user put in, not the canonical name?

**Comment by jensravesloot** [comment]
Could it also be related to this issue? 
[https://youtrack.jetbrains.com/issue/PY-52430/matplotlib-figures-get-frozen-when-executed-from-ipython-console](https://youtrack.jetbrains.com/issue/PY-52430/matplotlib-figures-get-frozen-when-executed-from-ipython-console)

**Comment by name-used** [comment]
> This could well be a duplicate of #28449, meaning that PyCharm need to support the changes in backend registration introduced in Matplotlib 3.9.0.

It seems that as described in this question, there will be an error when I use 'tkagg', but if I use the exact uppercase and lowercase letters' TkAgg ', there will be no error.
Thank you very much for providing the reference!

**Comment by name-used** [comment]
> Another point for measure: I can run the program on RHEL8 with PyCharm Community 2024.1.4 (using Matplotlib 3.9 from source, but I doubt it is related to that).
> 
> Just to be sure: the program runs OK from a Python shell? So it is only PyCharm that is the problem?

It would be all ok in pycharm python-run-mode. Only in "pycharm inline-mode", it would come to a bug.
The "pycharm inline-mode" includes:
1. When running with pycharm python-debug-mode and stop at any breakpoint, type codes in debug-console and run plt.show().
2. Open the pycharm-python-console. Directly type codes in pycharm-python-console and run plt.show().

**Comment by jensravesloot** [comment]
For the issue I am having I tried the following two pieces of code in the PyCharm console:
```
import matplotlib
matplotlib.use('TkAgg')
import matplotlib.pyplot as plt
print(matplotlib.get_backend())
plt.plot((1, 4, 6))
plt.show()
```

```
import matplotlib
matplotlib.use('tkagg')
import matplotlib.pyplot as plt
print(matplotlib.get_backend())
plt.plot((1, 4, 6))
plt.show()
```

Both cases yielded the same result, i.e., the plot still crashes.

**Comment by ianthomas23** [comment]
@jensravesloot Have you reported this to PyCharm? What response did you receive?

**Comment by jensravesloot** [comment]
> @jensravesloot Have you reported this to PyCharm? What response did you receive?

@ianthomas23, I think the issue mentioned here is the same: [https://youtrack.jetbrains.com/issue/PY-52430/matplotlib-figures-get-frozen-when-executed-from-ipython-console](https://youtrack.jetbrains.com/issue/PY-52430/matplotlib-figures-get-frozen-when-executed-from-ipython-console). I replied there mentioning this thread, but so far I don't have a useful reply. The more I try, the more I start to believe this is indeed a bug on PyCharm's side.

**Comment by ianthomas23** [comment]
> The more I try, the more I start to believe this is indeed a bug on PyCharm's side.

Indeed. PyCharm and similar need to support the changes made in matplotlib#27948, in particular the first item about backend names being case-insensitive. Until that happens Matplotlib code that is valid in say JupyterLab will not necessarily be valid in PyCharm.

I think we should close this issue as it is really a duplicate of #28449
